src/app/systelab-components/contextmenu/context-menu.component.ts

```diff
@@ -1,45 +1,71 @@
-import {Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output, Renderer2, ViewChild} from '@angular/core';
-import {ContextMenuActionData} from './context-menu-action-data';
-import {ContextMenuOption} from './context-menu-option';
+import {
+	ChangeDetectorRef,
+	Component,
+	ElementRef,
+	EventEmitter, HostListener,
+	Input,
+	OnDestroy,
+	OnInit,
+	Output,
+	QueryList,
+	Renderer2,
+	ViewChild,
+	ViewChildren
+} from '@angular/core';
+import { ContextMenuActionData } from './context-menu-action-data';
+import { ContextMenuOption } from './context-menu-option';
 
 declare var jQuery: any;
 
 @Component({
-	selector: 'systelab-context-menu',
+	selector:    'systelab-context-menu',
 	templateUrl: 'context-menu.component.html',
 })
 export class ContextMenuComponent implements OnInit, OnDestroy {
 
 	@ViewChild('dropdownparent') public dropdownParent: ElementRef;
 	@ViewChild('dropdownmenu') public dropdownMenuElement: ElementRef;
+	@ViewChildren('childdropdownmenu') public childDropdownMenuElement: QueryList<ElementRef>;
 	@ViewChild('dropdown') public dropdownElement: ElementRef;
+	@ViewChild('scrollableList') public scrollableList: ElementRef;
 
 	@Output() public action = new EventEmitter();
 
 	@Input() public contextMenuOptions: Array<ContextMenuOption>;
 
-	@Input() public elementID: string;
+	@Input() public elementID = (Math.floor(Math.random() * (999999999999 - 1))).toString();

```

Comment:
It's not a good practice to set an elementID using the random function.

Reply:
The elementID can be specified as input. In case is not set, a random is created just to avoid duplication problems.